docs/dev-guides/8-xrpackage-engine-api.md

```diff
@@ -39,57 +97,182 @@ The `XRPackage` API is documented [in the previous section](./7-xrpackage-api.md
+
+**Returns**: Nothing
 
 ## `packageRequestAnimationFrame`
 
-## `packageRequestPresent`
+## `packageRequestPresent(p)`
+
+_Marks the given package as its `onrequestpresent` WebXR event being received._
+
+**Parameters**: `p` is the `XRPackage` that has received the `onrequestpresent` event.
+
+**Returns**: an `Object` with key `session` that is this `XRPackageEngine` instance's session.
+
+## `remove(p, reason)`
+
+_Removes an XRPackage from the Engine._
+
+**Parameters**:
+
+- `p` is the `XRPackage` to remove
+- `reason` is an optional string describing why the package is being removed
+
+**Returns**: Nothing
+
+**Throws** an `Error` if the given `p` is not a child of this `XRPackageEngine` instance.
 
 ## `render`
 
-## `reset`
+## `reset()`
+
+_Removes all the children of the XRPackageEngine instance._
+
+**Parameters**: None
+
+**Returns**: Nothing
+
+## `resize(width, height, devicePixelRatio)`
+
+_Resizes this XRPackageEngine instance._
 
-## `resize`
+**Parameters**: all parameters are optional, with defaults as follows:
 
-## `setCamera`
+| Key                | Default                         | Description                         |
+| ------------------ | ------------------------------- | ----------------------------------- |
+| `width`            | `this.options.width`            | The new width of the renderer       |
+| `height`           | `this.options.height`           | The new height of the renderer      |
+| `devicePixelRatio` | `this.options.devicePixelRatio` | The new pixel ratio of the renderer |
+
+**Returns**: Nothing |
+
+## `setCamera(camera)`
+
+**Parameters**: `camera` is the camera to set for the Engine (e.g. a `Three.PerspectiveCamera`)
+
+**Returns**: Nothing
 
 ## `setClearFreeFramebuffer`
 
-## `setEnv`
+## `setEnv(key, value)`
 
-## `setGamepadsConnected`
+_Sets a single key-value environment pair._
 
-## `setMatrix`
+**Parameters**: a `key` and `value` to be associated with that `key`.
 
-## `setMicrophoneMediaStream`
+**Returns**: Nothing
 
-## `setRigMatrix`
+## `setGamepadsConnected(connected)`
 
-## `setScale`
+**Parameters**: `connected` is a `Boolean` representing whether the gamepads are now connected.
 
-## `setSession`
+**Returns**: Nothing
 
-## `setXrFramebuffer`
+## `setMatrix(m)`
 
-## `start`
+_Sets the matrix for this `XRPackageEngine` and sets all the children to need their matrix world updating._
 
-## `tick`
+**Parameters**: `m` is the matrix to set.
+
+**Returns**: Nothing
+
+## `setMicrophoneMediaStream(mediaStream)`
+
+**Parameters**: `mediaStream` is the media stream to set for this `XRPackageEngine`.
+
+**Returns**: Nothing
+
+**Example**:
+
+```js
+const pe = new XRPackageEngine();
+const mediaStream = await navigator.mediaDevices.getUserMedia({
+  audio: true,
+});
+pe.setMicrophoneMediaStream(mediaStream);
+```
+
+## `setRigMatrix(rigMatrix)`
+
+**Parameters**: the `rigMatrix` to set. If this is falsy, the rig matrix is disabled.
```

Comment:
Should note that the rig matrix is the offset of the user's avatar from the scene camera, which is useful for implementing third-person view modes.